x-pack/plugins/watcher/public/sections/watch_status/components/watch_detail.tsx

```diff
@@ -58,10 +61,74 @@ const WatchDetailUi = ({ intl, watchId }: { intl: InjectedIntl; watchId: string
         );
       },
     },
+    {
+      actions: [
+        {
+          render: (action: any) => {
+            if (action.isAckable) {
+              return (
+                <EuiToolTip
+                  content={i18n.translate(
+                    'xpack.watcher.sections.watchDetail.watchTable.ackActionCellTooltipTitle',
+                    {
+                      defaultMessage: 'Acknowledge this watch action',
+                    }
+                  )}
+                >
+                  <EuiButtonEmpty
+                    iconType="check"
+                    isLoading={isActionStatusLoading}
+                    onClick={async () => {
+                      setIsActionStatusLoading(true);
+                      try {
+                        const watchStatus = await ackWatchAction(watchDetail.id, action.id);
+                        setIsActionStatusLoading(false);
+                        return setActionStatuses(watchStatus.actionStatuses);
+                      } catch (e) {
+                        setIsActionStatusLoading(false);
+                        toastNotifications.addDanger(
+                          i18n.translate(
+                            'xpack.watcher.sections.watchDetail.watchTable.ackActionErrorMessage',
+                            {
+                              defaultMessage: 'Error acknowledging action {actionId}',
+                              values: {
+                                actionId: action.id,
+                              },
+                            }
+                          )
+                        );
+                      }
+                    }}
+                  >
+                    <FormattedMessage
+                      id="xpack.watcher.sections.watchDetail.watchTable.ackActionCellTitle"
+                      defaultMessage="Acknowledge"
+                    />
+                  </EuiButtonEmpty>
+                </EuiToolTip>
+              );
+            }
+            return <Fragment />;
+          },
+        },
+      ],
+    },
   ];
 
   const { error, data: watchDetail, isLoading } = loadWatchDetail(watchId);
 
+  const [actionStatuses, setActionStatuses] = useState<any[]>([]);
+  const [isActionStatusLoading, setIsActionStatusLoading] = useState<boolean>(false);
```

Comment:
I think it'd be a little easier to follow if the variables used in the `columns` array were defined before `columns` is defined. At the same time, I think it makes sense to keep the hooks grouped together. How would you feel about moving all of the hooks to line 27, before `pagination` and `columns` are defined?

```js
  const { error, data: watchDetail, isLoading } = loadWatchDetail(watchId);

  const [actionStatuses, setActionStatuses] = useState<any[]>([]);
  const [isActionStatusLoading, setIsActionStatusLoading] = useState<boolean>(false);

  useEffect(
    () => {
      if (watchDetail) {
        setActionStatuses(watchDetail.watchStatus.actionStatuses);
      }
    },
    [watchDetail]
  );
```

Reply:
yeah, that makes sense. fixed.